module-07-pulumi-esc/exercise-02-aws-secrets-manager.md

```diff
@@ -0,0 +1,104 @@
+# Exercise 02: AWS Secrets Manager and Importing Environments
+
+In this exercise, you'll see how ESC integrates with popular secrets managers like AWS Secrets Manager. You can use ESC to pull dynamic secrets from a range of secret sources. For a complete list of services ESC integrates with, see [Pulumi ESC Integrations](https://www.pulumi.com/docs/esc/integrations/).
+
+You will build a Lambda function that queries the GitHub API.
+
+NOTE: The Lambda function is written in NodeJS as NodeJS Lambda functions are significantly easier to deploy compared to Python. You will need to install NodeJS in order to complete these steps.
+
+## Setup Steps
+
+The `setup` stack will take a GitHub token, place it in AWS Secrets Manager, and generate an ESC environment that reads the secret.
+
+1. Create a [GitHub personal token](https://github.com/settings/tokens?type=beta). The token can be read-only.
+1. Add the GitHub token as a secret config value to the setup infra for this exercise, e.g.:
+
+    ```bash
+    cd exercise-02-aws-secrets-manager
+    cd setup
+    pulumi config set githubToken github_pat_11AADEH5Q0MJ9dUZbsyruI_n49twNSe6zTIKmCNwZ5458492eugfiohdjksRL7ZyDGNLVQQR1ZfQcSxj --secret 
+    ```
+
+1. (Optional) If you want to use ESC OIDC credentials, add them to this stack:
+
+    ```bash
+    pulumi config env add aws/aws-oidc-admin
+    ```
+
+1. Deploy the infrastructure:
+
+    ```bash
+    npm i
+    pulumi up
+    ```
+
+    Note that if you are working from a shared Pulumi organization, you may need to first create a new stack:
+
+    ```bash
+    pulumi stack init <your-name>
+    ```
+
+1. When the stack has deployed, note the command to add an environment to the stack containing the Lambda function.
+
+    ```bash
+    pulumi stack output command
+    ```
+
+## Exercise Steps
+
+1. Check out the generated ESC environment in the Pulumi Cloud console. Note that it reuses the OIDC credentials we set up earlier. These OIDC credentials are the role that Pulumi Cloud will use to access the secrets in AWS Secrets Manager.
+
+1. Install the Lambda function's dependencies:
+
+    ```bash
+    cd ../exercise/lambda
+    npm i
+    ```
+
+1. Add the environment to the exercise stack. Note that because the environment imports the OIDC credentials we used earlier, we do not need to add the OIDC environment separately.
```

Comment:
```suggestion
1. Add the environment to the exercise stack via executing the command from the last step of the Setup task above
```

(in case @smithrobs didn't know about this ^^)